base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/generation/GeneratorFacade.java

```diff
@@ -0,0 +1,118 @@
+package uk.ac.stfc.isis.ibex.scriptgenerator.generation;
+
+import uk.ac.stfc.isis.ibex.scriptgenerator.Activator;
+import uk.ac.stfc.isis.ibex.scriptgenerator.generation.GeneratedLanguage;
+import uk.ac.stfc.isis.ibex.scriptgenerator.generation.GeneratorContext;
+import uk.ac.stfc.isis.ibex.scriptgenerator.generation.InvalidParamsException;
+import uk.ac.stfc.isis.ibex.scriptgenerator.generation.UnsupportedLanguageException;
+import uk.ac.stfc.isis.ibex.scriptgenerator.pythoninterface.Config;
+import uk.ac.stfc.isis.ibex.scriptgenerator.table.ActionsTable;
+import uk.ac.stfc.isis.ibex.logger.IsisLog;
+
+import java.util.HashMap;
+
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Class to use to generate a script or check parameters are valid in a specified language.
+ * 
+ * @author James King
+ *
+ */
+public class GeneratorFacade {
+	
+	private static final Logger LOG = IsisLog.getLogger(GeneratorFacade.class);
+	
+	/**
+	 * Part of the strategy design pattern to generate based on which language to use.
+	 */
+	private static final GeneratorContext generatorContext = new GeneratorContext(Activator.getModel().getPythonInterface());	
+	
+	/**
+	 * Generate a python script from the actionsTable passed.
+	 * 
+	 * @param actionsTable The script generator contents to generate the script from.
+	 * @param config The instrument config to generate the script with.
+	 * @return The path to the generated script.
+	 * @throws InvalidParamsException Thrown if the values for the parameters in the actionTable are invalid.
+	 */
+	public static String generate(ActionsTable actionsTable, Config config) throws InvalidParamsException {
+		try {
+			return generate(actionsTable, config, GeneratedLanguage.PYTHON);
+		} catch(UnsupportedLanguageException e) {
+			LOG.error("Unsupported language when trying to generate script: " + e);
+			return null;
```

Comment:
Yes I agree. The reason I did this is because we always expect Python to be a supported language, however the default may change and could introduce bugs. I have introduced a mechanism to throw this back up to the GUI which will display an error message. I have also added tests to check that this default language method does not throw an exception, because we shouldn't be doing that and tests to check if there is an unsupported language chosen then it is caught.

Reply:
The GUI also suspends checking parameter validity if the language is not supported. This can be tested by inserting the GeneratedLanguage.UNSUPPORTED_LANGUAGE where appropriate when checking validity errors.